src/controller/src/lib.rs

```diff
@@ -213,6 +213,14 @@ where
             }
         }
     }
+
+    /// Produces a timestamp that reflects all data available in
+    /// `source_ids` at the time of the function call.
+    #[allow(unused)]
+    #[allow(clippy::unused_async)]
+    pub async fn recent_timestamp(&self, source_ids: Vec<GlobalId>) -> T {
```

Comment:
Hard to say at the moment, but this probably can take a slice instead of a vec?

Reply:
Just switched to a slice.